syntax = "proto3";

package GM;

// [请求]获取公告
message PROBUFF_GM_GET_NOTICE
{
};

message gm_notice
{
	optional int64	m_id = 1;// 公告id
	optional string	m_notice = 2;
	optional int32	m_starttime = 3;// 开始时间
	optional int32	m_finishtime = 4;// 结束时间
};

// [回复]获取公告
message PROBUFF_GM_GET_NOTICE_RESPONSE
{
	repeated gm_notice m_notice = 3;
};


// [请求]添加公告
message PROBUFF_GM_ADD_NOTICE
{
	optional gm_notice m_notice = 3;
};

// [回复]添加公告
message PROBUFF_GM_ADD_NOTICE_RESPONSE
{
	optional bool m_stat = 3;
};

// [请求]删除公告
message PROBUFF_GM_DEL_NOTICE
{
	optional int64 m_id = 3;
};

// [回复]删除公告
message PROBUFF_GM_DEL_NOTICE_RESPONSE
{
	optional bool m_stat = 3;
};

// [请求]PHP回调发货
message PROBUFF_GM_RECHARGE
{
	optional int64 m_rechargeid = 1;
	optional int64 m_roleid = 2;
	optional string m_orderid = 3;
};

// [回复]PHP回调发货
message PROBUFF_GM_RECHARGE_RESPONSE
{
	optional bool m_rechargeid = 1;
	optional int64 m_roleid = 2;
	optional string m_orderid = 3;
	optional int32 m_stat = 4;
};